packages/paypal-commerce-integration/src/paypal-commerce-fastlane/paypal-commerce-fastlane-customer-strategy.ts

```diff
@@ -175,16 +175,24 @@ export default class PayPalCommerceFastlaneCustomerStrategy implements CustomerS
         }
 
         if (shippingAddress && cart.lineItems.physicalItems.length > 0) {
-            const consignments = state.getConsignments() || [];
+            await this.paymentIntegrationService.updateShippingAddress(shippingAddress);
+
+            const updatedState = this.paymentIntegrationService.getState();
+            const paymentMethod =
+                updatedState.getPaymentMethodOrThrow<PayPalCommerceInitializationData>(methodId);
+            const { isEnableFastlaneShippingOptionAutoSelect } =
+                paymentMethod.initializationData || {};
+            const consignments = updatedState.getConsignments() || [];
             const availableShippingOptions = consignments[0]?.availableShippingOptions || [];
             const firstShippingOption = availableShippingOptions[0];
             const recommendedShippingOption = availableShippingOptions.find(
                 (option) => option.isRecommended,
             );
 
-            await this.paymentIntegrationService.updateShippingAddress(shippingAddress);
-
-            if (recommendedShippingOption || firstShippingOption) {
+            if (
+                (recommendedShippingOption || firstShippingOption) &&
+                (isEnableFastlaneShippingOptionAutoSelect ?? true)
```

Comment:
It is not necessary to set `?? true` here, since if the experiment is disabled then this variable will be `true` by default

Reply:
removed